src/fsharp/FSharp.Core/tasks.fs

```diff
@@ -0,0 +1,391 @@
+// TaskBuilder.fs - TPL task computation expressions for F#
+//
+// Originally written in 2016 by Robert Peele (humbobst@gmail.com)
+// New operator-based overload resolution for F# 4.0 compatibility by Gustavo Leon in 2018.
+// Revised for insertion into FSHarp.Core by Microsoft, 2019.
+//
+// Original notice:
+// To the extent possible under law, the author(s) have dedicated all copyright and related and neighboring rights
+// to this software to the public domain worldwide. This software is distributed without any warranty.
+//
+// Updates:
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.FSharp.Core.CompilerServices
+    /// A marker interface to give priority to different available overloads
+    type IPriority3 = interface end
```

Comment:
Just curious, why an interface in lieu of a class?
In F#+ I use `... class end`.
Then in Witnesses I would just inherit from Priority3.

Reply:
No great preference either way